lib/Spot/Entity.php

```diff
@@ -137,6 +139,14 @@ public function data($data = null, $modified = true)
                 $v = $this->__get($k, $v);
             }
 
+            foreach (self::$relationFields[$entityName] as $relationField) {
+                $relation = $this->relation($relationField);
+
+                if ($relation instanceof Entity\Collection) {
```

Comment:
You should also check for `instanceof Entity` for single-record relations, and add a test for a HasOne or BelongsTo relationship.